src/RA_Defs.h

```diff
@@ -50,25 +50,19 @@
 #define NODEFERWINDOWPOS
 #define NOMCX  
 
-
 #include <Windows.h>
 #include <WindowsX.h>
 #include <ShlObj.h>
 #include <tchar.h>
 
-#ifdef WIN32_LEAN_AND_MEAN
-#include <MMSystem.h>
-#include <ShellAPI.h>
-#include <CommDlg.h>
-#endif // WIN32_LEAN_AND_MEAN
-
 #include <sstream>
+#include <string_view> 
```

Comment:
compile time string building you can't modify it though. This one requires C++17. I noticed a lot of desire for compile-time stuff so C++17 should actually be a requirement because of its vast compile-time support via constant expressions.